src/mapgen.cpp

```diff
@@ -259,28 +259,36 @@ static tripoint_bub_ms get_point_from_direction( int direction,
 }
 
 static bool tile_can_have_blood( map &md, const tripoint_bub_ms &current_tile,
-                                 bool wall_streak )
+                                 bool wall_streak, int days_since_cataclysm )
 {
     // Wall streaks stick to walls, like blood was splattered against the surface
     // Floor streaks avoid obstacles to look more like a person left the streak behind (Not walking through walls, closed doors, windows, or furniture)
     if( wall_streak ) {
-        return  md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_WALL, current_tile );
+        return md.has_flag_ter( ter_furn_flag::TFLAG_WALL, current_tile ) &&
+               !md.has_flag_ter( ter_furn_flag::TFLAG_NATURAL_UNDERGROUND, current_tile );
     } else {
-        return !md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_WALL, current_tile ) &&
-               !md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_WINDOW, current_tile ) &&
-               !md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_DOOR, current_tile ) &&
+        if( !md.has_flag_ter( ter_furn_flag::TFLAG_INDOORS, current_tile ) &&
+            !x_in_y( ( 30.0 - days_since_cataclysm ) / 30.0, 1 ) ) {
```

Comment:
you can simplify this a bit by removing the double-negative
```suggestion
            x_in_y( days_since_cataclysm / 30.0, 1 ) ) {
```

(you can also change that further to `x_in_y( days_since_cataclysm, 30.0 )` but idk if that looks better)